packages/flutter_tools/lib/src/android/gradle_errors.dart

```diff
@@ -664,3 +647,23 @@ final GradleHandledError incompatibleCompileSdk35AndAgpVersionHandler = GradleHa
   },
   eventLabel: 'incompatible-compile-sdk-and-agp',
 );
+
+@visibleForTesting
+final GradleHandledError r8DexingBugInAgp73Handler = GradleHandledError(
+  test: (String line) => line.contains('com.android.tools.r8.internal.Y10: Unused argument with users'),
```

Comment:
You may not want to match `Y10` as that identifier could be different between minor versions of R8 (it's very likely a obfuscated class name)